libtenzir/src/index.cpp

```diff
@@ -534,6 +534,39 @@ caf::error index_state::load_from_disk() {
         = *ps_flatbuffer->partition_synopsis_as_legacy();
       if (auto error = unpack(synopsis_legacy, ps.unshared()))
         return error;
+      // Add partition file sizes.
+      {
+        uint64_t bitmap_file_size = std::filesystem::file_size(part_path, err);
+        if (err) {
+          TENZIR_WARN("failed to get the size of the partition index file at "
+                      "{}: {}",
+                      part_path, err.message());
+          bitmap_file_size = 0u;
+        }
+        ps.unshared().indexes_file = {
+          .url = fmt::format("file://{}", canonical(part_path)),
+          .size = bitmap_file_size,
+        };
+        ps.unshared().sketches_file = {
+          .url = fmt::format("file://{}", canonical(synopsis_path)),
+          .size = chunk->get()->size(),
+        };
+        // We just assume the store file path, currently feather is the only
+        // existing implementation.
+        auto store_path
+          = dir / ".." / "archive" / fmt::format("{}.feather", partition_uuid);
```

Comment:
```suggestion
        auto store_path
          = dir / ".." / "archive" / fmt::format("{}.{}", partition_uuid, store_backend);
```

Reply:
The `store_backend` gets passed to the index on startup, we also need to persist it in the index state. Note, however, that this would assume that the store backend never changes over time—the actually correct approach would be to get the path from the stores themselves.

Reply:
> the actually correct approach would be to get the path from the stores themselves.

I think you mean the store header from the partition, but opening that and reading from all partition files would slow down the index startup significantly.

Reply:
I went with a slightly more robust approach of listing the contents of listing the `archive` directory. This does not break when the user changes the `store-backend`.
Please resolve if you're d'accord.